tests/spread/plugins/craft-parts/matter-sdk/snapcraft.yaml

```diff
@@ -0,0 +1,48 @@
+name: matter-lighting
+summary: Matter plugin test
+description: An lighting application to test the matter plugin.
+version: "1.0"
+
+base: core22
+
+grade: stable
+build-base: core22
+confinement: strict
+
+layout:
+  /mnt:
+    bind: $SNAP_COMMON/mnt
+
+apps:
+  matter-lighting:
+    daemon: simple
+    command: bin/lighting-app
+    install-mode: disable
+    plugs:
+      - network
+      - network-bind
+      - bluez
+      - avahi-control
+
+parts:
+  matter-sdk:
+    plugin: matter-sdk
+    matter-sdk-version: "1536ca20c5917578ca40ce509400e97b52751788" # use this commit with ptpython version fix; needs to be updated once matter sdk have a stable release
+
+  lighting:
+    plugin: nil
+    after: [matter-sdk]
+    override-build: |
+      # Source the Matter SDK environment variables
+      source ../../matter-sdk/build/matter-sdk-env.sh
```

Comment:
this is fragile and probably hard to discover, a reason to perhaps use an extension instead

Reply:
For this particular command, we could use an extension instead. As for the Matter SDK plugin itself, the reason we chose a plugin over an extension is to provide the option for the user to specify the Matter SDK version.